src/middleware/persist.ts

```diff
@@ -19,10 +19,20 @@ export interface PersistStorage<S> {
   removeItem: (name: string) => void | Promise<void>
 }
 
-export function createJSONStorage<S>(
-  getStorage: () => StateStorage
+
+export interface PersistStorageOptions <S, T = string> {
+  serialize?: (value:StorageValue<S> ) => T 
+  deserialize?: (value: T | null) => StorageValue<S>
+}
+
+export function createJSONStorage<S, T = string>(
```

Comment:
Separating `createJSONStorage` is exactly the reason for this kind of use cases.

The point of `createJSONStorage` is for JSON.parse/stringify.
I would recommend creating a third-party library for broader use cases.